### PR TITLE
KUZBO-131 treeview refactor

### PR DESCRIPTION
**Pull request by xbill82**
Better Treeview component structure for testing.

### CODE REVIEW

File: src/components/Data/Browse/IndexBranch.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+  <div v-bind:class="{ 'open': open }">
+    <i v-if="collectionCount(index)" class="fa fa-caret-right tree-toggle" aria-hidden="true" @click="toggleBranch()"></i>
+    <!-- v-link="{name: 'DataIndex', params: {index: index.name}}" -->
+    <a class="tree-item truncate"
+       v-bind:class="{ 'active': isIndexActive($route, index.name) }">
+      <i class="fa fa-database" aria-hidden="true"></i>
+      <strong>{{index.name}}</strong> ({{collectionCount(index)}})
+    </a>
+    <ul class="collections">
+      <li v-for="collection in index.collections.stored">
```
Comment (stafyniaksacha) [inline src/components/Data/Browse/IndexBranch.vue]: Hum, why we have stored collection in top of realtime ones ? 
I think we have to compute a list containing all collection, that permit to apply a sort order.

File: src/components/Data/Browse/IndexBranch.vue [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+        count += index.collections.stored.length
+      }
+
+      if (index.collections.realtime) {
+        count += index.collections.realtime.length
+      }
+
+      return count
+    },
+    isIndexActive (routeObject, indexName) {
```
Comment (stafyniaksacha) [inline src/components/Data/Browse/IndexBranch.vue]: I think we have to do other checks here:
 1. if the index or collection is selected, open the tab (that's ok)
 2. if the user has open or close a section we have to let into this state

File: src/components/Data/Browse/IndexBranch.vue [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      return routeObject &&
+             routeObject.params &&
+             routeObject.params.index === indexName &&
+             !routeObject.params.collection
+    },
+    isCollectionActive (routeObject, collectionName) {
```
Comment (stafyniaksacha) [inline src/components/Data/Browse/IndexBranch.vue]: same as above

File: src/components/Data/Browse/IndexBranch.vue (hunk 1 above)
Comment (xbill82) [inline src/components/Data/Browse/IndexBranch.vue]: good point

**Comment by stafyniaksacha** [comment]
apart my feedback, it's ok to me

File: src/components/Data/Browse/IndexBranch.vue (hunk 2 above)
Comment (xbill82) [inline src/components/Data/Browse/IndexBranch.vue]: Hmmm I don't get the point.
This is just a helper to tell whether the given index name matches the index of the route. It has no impact on the branch collapsing.

File: src/components/Data/Browse/IndexBranch.vue (hunk 1 above)
Comment (AnthonySendra) [inline src/components/Data/Browse/IndexBranch.vue]: :+1: for let like this and discuss later if we'll think there is a problem

File: src/components/Data/Browse/IndexBranch.vue [hunk 4]
```diff
@@ -0,0 +1,166 @@
+<template>
+  <div v-bind:class="{ 'open': open }">
+    <i v-if="collectionCount(index)" class="fa fa-caret-right tree-toggle" aria-hidden="true" @click="toggleBranch()"></i>
+    <!-- v-link="{name: 'DataIndex', params: {index: index.name}}" -->
+    <a class="tree-item truncate"
+       v-bind:class="{ 'active': isIndexActive($route, index.name) }">
```
Comment (AnthonySendra) [inline src/components/Data/Browse/IndexBranch.vue]: Could we say we don't use v-bind: and v-on: but shortcut instead? like :class? (just like for @click in line 3)

**Comment by AnthonySendra** [comment]
despite my feedback :+1:

File: src/components/Data/Browse/IndexBranch.vue (hunk 1 above)
Comment (stafyniaksacha) [inline src/components/Data/Browse/IndexBranch.vue]: ok !

File: src/components/Data/Browse/IndexBranch.vue (hunk 4 above)
Comment (xbill82) [inline src/components/Data/Browse/IndexBranch.vue]: done

**Pull request merged**